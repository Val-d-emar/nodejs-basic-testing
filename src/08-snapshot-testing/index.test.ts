// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    // Write your test here
    const tree = generateLinkedList(['http://ya.ru']);
    expect(tree).toMatchSnapshot();
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    // Write your test here
    const tree = generateLinkedList([
      'http://ya.ru',
      'https://jsonplaceholder.typicode.com',
    ]);
    expect(tree).toMatchSnapshot();
  });
});
