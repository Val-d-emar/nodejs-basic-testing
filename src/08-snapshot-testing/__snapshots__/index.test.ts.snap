// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 1 1`] = `
{
  "next": {
    "next": null,
    "value": null,
  },
  "value": "http://ya.ru",
}
`;

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": null,
      "value": null,
    },
    "value": "https://jsonplaceholder.typicode.com",
  },
  "value": "http://ya.ru",
}
`;
